mantidproject/mantid
# Chained BinMD slice viewer issues.

**To Test**
See the issue description for more detail.

 A useful script was created by Owen which will generate some data, setup, and show the slice viewer:

``` python
# Lets create a fake MDWorkspace
mdworkspace = CreateMDWorkspace(Dimensions=3, Extents='-10,10,-10,10,-10,10', Names='H,K,L', Units='U,U,U')
# Add A Peak at H=0, K=0, L=1
FakeMDEventData(InputWorkspace=mdworkspace, PeakParams='100000,0,0,0,1')
# Add another peak at H=2, K=2, L=0
FakeMDEventData(InputWorkspace=mdworkspace, PeakParams='100000,2,2,0,1')
# Open SliceViewer and set the slice point to K=1, half-way between the two Peaks.
plotSlice(mdworkspace, xydim=['H','L'], slicepoint=[0,1,0]) 
```

From there set the following options:
- Toggle robin mode on
- Make sure that H is the X and L is the Y so we're slicing on K
- Slice point to 1.0
- Thickness to 1.0

Now right click on the newly generated workspace (called mdworkspace_rebinned) and open the slice viewer and toggle dynamic rebin mode. The screen should look something like the following:

<img width="635" alt="screen shot 2016-07-19 at 12 17 18" src="https://cloud.githubusercontent.com/assets/2487781/16948139/dc1527ac-4daa-11e6-80fb-741469175299.png">

Note that two peaks are visible and that the bin size of the K axis is 1.

Fixes #16860.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
